### PR TITLE
add custom error check when many rows are entirely missing

### PR DESCRIPTION
**Pull request by vandenman**
@TimKDJ, do you think this should be a common check or is it so specific that it's only necessary for network analysis?

### CODE REVIEW

**Comment by TimKDJ** [comment]
Common check would be good!

**Comment by TimKDJ** [comment]
Could you rebase this to see if it passes?

**Comment by vandenman** [comment]
why didn't I make this PR from my own fork? oh well...

**Comment by vandenman** [comment]
@TimKDJ should be good now! I can also remove the commented stuff if you prefer

**Comment by TimKDJ** [comment]
I think we can do without the comment, yes

**Comment by TimKDJ** [comment]
Is it easy to add a unit test that tests this error check?

**Comment by vandenman** [comment]
@TimKDJ I added a unit test!

File: tests/testthat/test-networkanalysis.R [hunk 1]
```diff
@@ -165,3 +165,22 @@ for (estimator in estimators) {
...
+dataset[sample(n, 0.5 * n), ] <- NA
+dataset <- as.data.frame(dataset)
+options <- jaspTools::analysisOptions("NetworkAnalysis")
+options$estimator <- "EBICglasso"
+options$variables <- c("V1", "V2", "V3")
+results <- jaspTools::runAnalysis("NetworkAnalysis", dataset, options)
+
+test_that("Too many missing rows returns an error", {
+  r <- results[["results"]]
+  expect_true(r[["error"]])
```
Comment (TimKDJ) [inline tests/testthat/test-networkanalysis.R:184]: Nitpick: I'm pretty sure both a `fatalError` (= analysis did not catch the error but `runJaspResults()` did) and `validationError` (= `.hasErrors()` exited the analysis when validation checks did not pass) set `r[["error"]]` to `TRUE`. It'd be better to verify it's indeed a `validationError` and the analysis didn't just crap out.

File: tests/testthat/test-networkanalysis.R [hunk 2]
```diff
@@ -165,3 +165,22 @@ for (estimator in estimators) {
...
+  expect_true(any(grepl("rows", r[["errorMessage"]], ignore.case = TRUE)), label = "Entirely missing rows check")
```
Comment (TimKDJ) [inline tests/testthat/test-networkanalysis.R:185]: Although this would probably be the fail safe in that scenario ;)

**Review by TimKDJ — APPROVED** [review]

**Pull request merged**